server/Logic/HutelDate.cs

```diff
@@ -0,0 +1,62 @@
+using System;
+using System.Globalization;
+
+namespace hutel.Logic
+{
+    public class HutelDate
+    {
+        private DateTime DateTime { get; }
+
+        private const string _format = "yyyy-MM-dd";
+
+        public HutelDate(string date)
+        {
+            DateTime = DateTime.ParseExact(date, _format, CultureInfo.InvariantCulture);
```

Comment:
try:  "date": "2017-09-31"
dont throw even general exception,
so BadRequest answer is not formed.
I am not sure if this is expected behaviour